### ISSUE TITLE
Via Stations 

### ISSUE THREAD

**Issue by mattmoo44** [peter-mount/departureboards#4]
Via Stations are listed incorrectly at MTG, Mottingham for London Cannon Street to London Cannon Street loop services. 

The service is listed correctly as a Woolwich Arsenal train, but the via points are set for the whole journey back to London Cannon Street. 

Example:
DESTINATION: Woolwich Arsenal
VIA: Sidcup & Woolwich Arsenal
Calling at: New Eltham (14:37) Sidcup (14:40) Albany Park (14:43) Bexley (14:45) Crayford (14:51) Slade Green (14:59) Erith (15:02) Belvedere (15:04) Abbey Wood (15:08) Plumstead (15:11) Woolwich Arsenal (15:14) Woolwich Dockyard (15:16) Charlton (15:20) Westcombe Park (15:22) Maze Hill (15:24) Greenwich (15:27) Deptford (15:29) London Bridge (15:37) London Cannon Street (15:41)
Southeastern service. Last report: New Cross 14:18

**Comment by peter-mount** [comment]
Ok found it, RID 201902088702461

I can see for that service at MOTNGHM it's got the false Destination set for WOLWCHA but when calling points are generated they don't use the False Destination as the end point but show the rest of the schedule. Thinking about it, I think these should stop a the false destination if set.

I'm preparing the next version of the backend for when they finally upgrade the open version of Darwin so I'll see if I can get this into that

[201902088702461.txt](https://github.com/peter-mount/departureboards/files/2845495/201902088702461.txt)

**Comment by mattmoo44** [comment]
Thanks for getting back to me so quickly, I don't pretend to understand how the data works. 

Intresting there are false destinations. I did wonder how they managed to show these trains as terminating, when they actually don't terminate until much later.

**Comment by peter-mount** [comment]
False destinations are there to stop confusion on the platforms. For example it would be confusing for someone at Cannon St to see a service for Cannon St being advertised, so they set the false destination to an intermediate stop for all stations up to that stop. In this instance once the train gets to Woolwich then it could then show CST again as by then it wouldn't be confusing to the passenger.

**Comment by mattmoo44** [comment]
Makes sense, I was actually at London Bridge last night and they did accidentally shown this service as a London Cannon Street service. 

Had a few people taking angry scenic trips around Kent I expect.

**Comment by peter-mount** [comment]
This is now working on the test environment:
![Screenshot_2019-04-25_18-23-03](https://user-images.githubusercontent.com/4427085/56755329-33464580-6787-11e9-95e7-e69b6a14cc03.png)

The service detail page still shows Cannon St as origin & destination which is correct for that view as it's for the entire service:
![Screenshot_2019-04-25_18-23-58](https://user-images.githubusercontent.com/4427085/56755416-64bf1100-6787-11e9-862e-7145fb7d46b0.png)

**Comment by mattmoo44** [comment]
It is still showing the wrong via points on my UK departure board. For the station MTG Mottingham.

It now shows the false destination as Woolwich Arsenal but still says via 'Sidcup and Woolwich Arsenal.

It should just say via Sidcup

**Comment by peter-mount** [comment]
Ok thanks I think I know where that one is.

**Comment by mattmoo44** [comment]
I've got an additional update, I've realised the calling points are still for the whole journey back to Cannon Street as well.

**Comment by peter-mount** [comment]
I'll check, I think I have fixed it for calling points but not vias. I've
got some work to do with those with associated services anyhow so I'll fix
this at the same time.
